pkg/router/routeapihelpers/validation_test.go

```diff
@@ -1943,6 +2095,71 @@ func TestExtendedValidateRoute(t *testing.T) {
 			},
 			expectedErrors: 1,
 		},
+		{
+			name: "Edge termination with self-signed cert using SHA1 with RSA key",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationEdge,
+						Certificate: testCertificateRsaSha1SelfSigned,
+						Key:         testCertificateRsaSha1SelfSignedKey,
+					},
+				},
+			},
+			expectedErrors: 0,
+		},
+		{
+			name: "Reencrypt termination with destination CA root and intermediate cert using SHA1 with RSA key",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination:              routev1.TLSTerminationReencrypt,
+						DestinationCACertificate: testCertificateRsaSha1CA + testCertificateRsaSha1Intermediate,
+					},
+				},
+			},
+			expectedErrors: 1,
+		},
+		{
+			name: "Reencrypt termination with destination CA root cert using SHA1 with RSA key",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination:              routev1.TLSTerminationReencrypt,
+						DestinationCACertificate: testCertificateRsaSha1CA,
+					},
+				},
+			},
+			expectedErrors: 0,
+		},
+		{
+			name: "Edge termination with root CA cert using SHA1 and server cert using SHA256",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination:   routev1.TLSTerminationEdge,
+						CACertificate: testCertificateRsaSha1CA,
+						Certificate:   testCertificateRsaSha256,
+						Key:           testCertificateRsaSha256Key,
+					},
+				},
+			},
+			expectedErrors: 0,
```

Comment:
Can you add a comment about why SHA1 in the CACertificate is okay?  I thought there would be an error here.